Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -140,6 +140,9 @@ public enum CertificateVerification: Sendable {
     /// be checked to see if they are valid for the given hostname.
     case noHostnameVerification
 
+    /// Peer certificate is optional
+    case noHostnameNoPeerVerification
```

Comment:
Adding a new case to a public enum is an API breaking change because a user could switch over it and the compiler would now complain that the new case `.noHostnameNoPeerVerification` is missing.
We will need to think of some other way to make this configurable. Maybe deprecating the current enum and adding a new struct that is backed by an enum.